src/controllers/Social.storageWithdraw.test.ts

```diff
@@ -0,0 +1,222 @@
+import { Account, providers, transactions, utils } from 'near-api-js';
+//import { randomBytes } from 'node:crypto';
+import { ViewMethodEnum } from '@app/enums';
+
+// credentials
+import { account_id as socialContractAccountId } from '@test/credentials/localnet/social.test.near.json';
+
+// controllers
+import Social from './Social';
+
+// helpers
+import accountAccessKey, {
+  IAccessKeyResponse,
+} from '@test/helpers/accountAccessKey';
+import convertNEARToYoctoNEAR from '@app/utils/convertNEARToYoctoNEAR';
+import createEphemeralAccount from '@test/helpers/createEphemeralAccount';
+
+describe(`${Social.name}#storageWithdraw`, () => {
+  let keyPair: utils.KeyPairEd25519;
+  let signer: Account;
+  let signerAccessKeyResponse: IAccessKeyResponse;
+
+  beforeEach(async () => {
+    const result = await createEphemeralAccount(convertNEARToYoctoNEAR('100'));
+
+    keyPair = result.keyPair;
+    signer = result.account;
+  });
+
+  it('should withdraw storage given an amount', async () => {
+    // arrange
+    const client = new Social({
+      contractId: socialContractAccountId,
+    });
+    let resultBefore: Record<string, unknown>;
+    let resultAfter: Record<string, unknown>;
+    let transaction: transactions.Transaction;
+
+    signerAccessKeyResponse = await accountAccessKey(signer, keyPair.publicKey);
+
+    // act
+    let account_id = signer.accountId;
+
+    //Make deposit to the account
+    //2N deposit
+    let deposit = '2000000000000000000000000';
+    transaction = await client.storageDeposit({
+      blockHash: signerAccessKeyResponse.block_hash,
+      nonce: BigInt(signerAccessKeyResponse.nonce + 1),
+      publicKey: keyPair.publicKey,
+      signer,
+      account_id,
+      deposit,
+    });
+
+    // assert
+    // the transaction's actions should have `storage_deposit`
+    expect(transaction.actions).toHaveLength(1);
+
+    let [_, signedTransaction] = await transactions.signTransaction(
+      transaction,
+      signer.connection.signer,
+      signer.accountId,
+      signer.connection.networkId
+    );
+    let { status } =
+      await signer.connection.provider.sendTransaction(signedTransaction);
+    let failure = (status as providers.FinalExecutionStatus)?.Failure || null;
+
+    if (failure) {
+      throw new Error(`${failure.error_type}: ${failure.error_message}`);
+    }
+
+    resultBefore = await signer.viewFunction({
+      args: {
+        account_id: signer.accountId,
+      },
+      contractId: socialContractAccountId,
+      methodName: ViewMethodEnum.StorageBalanceOf,
+    });
```

Comment:
Should we expose the `storage_of` function? I know it is set to private, but maybe it can be public.

Reply:
I think we should make it public.